AsyncDisplayKit/Details/ASPINRemoteImageDownloader.m

```diff
@@ -53,23 +123,39 @@ - (void)fetchCachedImageWithURL:(NSURL *)URL
 
 - (void)clearFetchedImageFromCacheWithURL:(NSURL *)URL
 {
-  PINRemoteImageManager *manager = [PINRemoteImageManager sharedImageManager];
+  PINRemoteImageManager *manager = [self sharedPINRemoteImageManager];
   NSString *key = [manager cacheKeyForURL:URL processorKey:nil];
   [[[manager cache] memoryCache] removeObjectForKey:key];
 }
 
 - (nullable id)downloadImageWithURL:(NSURL *)URL
                       callbackQueue:(dispatch_queue_t)callbackQueue
-                   downloadProgress:(void (^)(CGFloat progress))downloadProgressBlock
-                         completion:(void (^)(UIImage *image, NSError * error, id downloadIdentifier))completion
+                   downloadProgress:(ASImageDownloaderProgress)downloadProgress
+                         completion:(ASImageDownloaderCompletion)completion;
 {
-  return [[PINRemoteImageManager sharedImageManager] downloadImageWithURL:URL options:PINRemoteImageManagerDownloadOptionsSkipDecode completion:^(PINRemoteImageManagerResult *result) {
+  return [[self sharedPINRemoteImageManager] downloadImageWithURL:URL options:PINRemoteImageManagerDownloadOptionsSkipDecode completion:^(PINRemoteImageManagerResult *result) {
     /// If we're targeting the main queue and we're on the main thread, complete immediately.
     if (ASDisplayNodeThreadIsMain() && callbackQueue == dispatch_get_main_queue()) {
-      completion(result.image, result.error, result.UUID);      
+#if PIN_ANIMATED_AVAILABLE
+      if (result.alternativeRepresentation) {
+        completion(result.alternativeRepresentation, result.error, result.UUID);
+      } else {
+        completion(result.image, result.error, result.UUID);
+      }
+#else
+      completion(result.image, result.error, result.UUID);
```

Comment:
I think you can have just one call to completion for all cases, and do:

`id <ASImageContainerProtocol> image = (PIN_ANIMATED_AVAILABLE ? result.alternativeRepresentation : result.image);`